src/core/api/chart_repository.go

```diff
@@ -305,6 +320,18 @@ func (cra *ChartRepositoryAPI) UploadChartVersion() {
 
 	// Directly proxy to the backend
 	chartController.ProxyTraffic(cra.Ctx.ResponseWriter, cra.Ctx.Request)
+
+	go func() {
```

Comment:
the trigger action should not happen here as it cannot guarantee success, you should move the trigger into the response handler.

Reply:
chart event related notification code deleted